sklearn/decomposition/pca.py

```diff
@@ -75,6 +80,8 @@ def _assess_dimension_(spectrum, rank, n_samples, n_features):
     spectrum_ = spectrum.copy()
     spectrum_[rank:n_features] = v
     for i in range(rank):
+        if spectrum_[i] < rcond:
```

Comment:
> we'll never see complex numbers will we?
PCA could be one of the algorithms that might with complex numbers. But we don't support it AFAIK.

We can either remove the abs everywhere, or add it also inside the `v` summation. Couldn't say which is better.